src/core/radio/ble_secure.cpp

```diff
@@ -74,7 +76,12 @@ Error BleSecure::Start(ConnectCallback aConnectHandler, ReceiveCallback aReceive
     mMtuSize = kInitialMtuSize;
 
     SuccessOrExit(error = otPlatBleEnable(&GetInstance()));
+
+    advertisementData = mTcatAgent.GetAdvertisementData(advertisementLen);
```

Comment:
Foreseeably users may try to initialize TCAT in the following order:
1) otBleSecureStart(...)
2) otBleSecureSetTcatVendorInfo(...)
3) otBleSecureTcatStart(...)

Because otBleSecureStart(...) already calls mTcatAgent.GetAdvertisementData() this would go wrong.

I suggest we should make sure otBleSecureStart(...) can be called without prior setting TCAT vendor data to enable BleSecure uses cases outside TCAT. If no advertisment data or TCAT vendor information was set before, only the Thread GATT service shall be advertised. Users may call otPlatBleGapAdvSetData(...) any time to set/update vendor specific advertisement data.
otBleSecureTcatStart(...) should itself set/update advertisement data by calling otPlatBleGapAdvSetData(...) using the latest set of TCAT vendor information.

Reply:
Are You sure that this correct way to do it?

In the current implementation user has to call:
1. otBleSecureStart(...)
2. otBleSecureTcatStart(...) with valid vendor info

The only change in this init process that we have now is that the vendor info is going to be verified earlier and BEL advertisement won't be started if vendor info is not set.